ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/MeteredLedgerDao.scala

```diff
@@ -0,0 +1,312 @@
+      recordTime: Instant,
+      offsetStep: OffsetStep,
+      reason: RejectionReason,
+  )(implicit loggingContext: LoggingContext): Future[PersistenceResponse] =
+    Timed.future(
+      metrics.daml.index.db.storeRejection,
+      ledgerDao.storeRejection(submitterInfo, recordTime, offsetStep, reason),
+    )
+
+  @nowarn
+  override def storeInitialState(
+      ledgerEntries: Vector[(Offset, LedgerEntry)],
+      newLedgerEnd: Offset,
+  )(implicit loggingContext: LoggingContext): Future[Unit] =
+    Timed.future(
+      metrics.daml.index.db.storeInitialState,
+      ledgerDao.storeInitialState(ledgerEntries, newLedgerEnd),
+    )
+
+  override def initializeLedger(ledgerId: LedgerId)(implicit
+      loggingContext: LoggingContext
+  ): Future[Unit] =
+    ledgerDao.initializeLedger(ledgerId)
+
+  override def initializeParticipantId(participantId: ParticipantId)(implicit
+      loggingContext: LoggingContext
+  ): Future[Unit] =
+    ledgerDao.initializeParticipantId(participantId)
+
+  override def reset()(implicit loggingContext: LoggingContext): Future[Unit] =
+    ledgerDao.reset()
+
+  override def storePartyEntry(
+      offsetStep: OffsetStep,
+      partyEntry: PartyLedgerEntry,
+  )(implicit loggingContext: LoggingContext): Future[PersistenceResponse] =
+    Timed.future(
+      metrics.daml.index.db.storePartyEntry,
+      ledgerDao.storePartyEntry(offsetStep, partyEntry),
+    )
+
+  override def storeConfigurationEntry(
+      offsetStep: OffsetStep,
+      recordTime: Instant,
+      submissionId: String,
+      configuration: Configuration,
+      rejectionReason: Option[String],
+  )(implicit loggingContext: LoggingContext): Future[PersistenceResponse] =
+    Timed.future(
+      metrics.daml.index.db.storeConfigurationEntry,
+      ledgerDao.storeConfigurationEntry(
+        offsetStep,
+        recordTime,
+        submissionId,
+        configuration,
+        rejectionReason,
+      ),
+    )
+
+  override def storePackageEntry(
+      offsetStep: OffsetStep,
+      packages: List[(Archive, PackageDetails)],
+      entry: Option[PackageLedgerEntry],
+  )(implicit loggingContext: LoggingContext): Future[PersistenceResponse] =
+    Timed.future(
+      metrics.daml.index.db.storePackageEntry,
+      ledgerDao.storePackageEntry(offsetStep, packages, entry),
+    )
+
+  override def storeTransactionEvents(preparedInsert: PreparedInsert)(implicit
+      loggingContext: LoggingContext
+  ): Future[PersistenceResponse] =
+    ledgerDao.storeTransactionEvents(preparedInsert)
+
+  override def completeTransaction(
+      submitterInfo: Option[SubmitterInfo],
+      transactionId: TransactionId,
+      recordTime: Instant,
+      offsetStep: OffsetStep,
+  )(implicit loggingContext: LoggingContext): Future[PersistenceResponse] =
+    ledgerDao.completeTransaction(submitterInfo, transactionId, recordTime, offsetStep)
+
+  // TODO append-only cleanup
+  override def storeTransactionState(preparedInsert: PreparedInsert)(implicit
```

Comment:
For ease of analysis, I would move this method to where the original was located.